### PR TITLE
[5.6] Set `relatedDependenciesBranch` to `release/5.6`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
### Motivation:

Tools depending on `release/5.6` branch of SwiftPM together with `release/5.6` branch of TSC fail to build due to conflicting dependencies.

### Modifications:

`relatedDependenciesBranch` now is set to `release/5.6` instead of `main` when SwiftPM `release/5.6` branch is checked out. This makes dependencies consistent and resolvable when other Swift tools are included in the dependency tree. 

### Result:

Tools depending on `release/5.6` branch of SwiftPM can be built without failures.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
Thanks @MaxDesiatov.  Also opened a corresponding on on `swift-driver` since local builds will fail otherwise:  https://github.com/apple/swift-driver/pull/929

**Comment by abertelrud** [comment]
Will mark approved once the swift-driver change is in so local desktop use doesn't break.  Thanks for this fix @MaxDesiatov!

**Comment by abertelrud** [comment]
Still waiting for there to be a 5.6 toolchain so that we can merge the SwiftDriver change.  Otherwise all non-CI builds of SwiftPM will start failing when we merge this.

**Comment by abertelrud** [comment]
Update: since there's no 5.6 toolchain yet, and therefore apple/swift-driver#929 is blocked from merging, I ran a cross-repo test using an empty-commit PR in the Swift compiler repo (https://github.com/apple/swift/pull/40613).  That seems to have succeeded, which means that we should be able to merge apple/swift-driver#929 soon and then this one after that.

**Comment by abertelrud** [comment]
Update: Still trying to get the swift-driver change in, otherwise desktop builds of SwiftPM will start failing once this is merged.

**Comment by abertelrud** [comment]
The SwiftDriver change has been merged, so I will do a local test to confirm that this is now fixed, and then merge it.

**Comment by abertelrud** [comment]
With the latest change to SwiftDriver (https://github.com/apple/swift-driver/pull/964), this looks good now for local builds as well.  I think we just need @tomerd's approval now for the 5.6 branch.

**Review by abertelrud — APPROVED** [review]

**Review by tomerd — APPROVED** [review]

**Pull request merged**
